pkg/keyvisual/input/api.go

```diff
@@ -118,9 +127,16 @@ func NewAPIPeriodicGetter(pdAddr string) regionpkg.RegionsInfoGenerator {
 	addr := fmt.Sprintf("%s/pd/api/v1/regions", pdAddr)
 	return func() (regionsInfo regionpkg.RegionsInfo, err error) {
 		resp, err := http.Get(addr) //nolint:bodyclose,gosec
-		if err == nil {
-			return read(resp.Body)
+		if err != nil {
+			return nil, err
+		} else if resp.StatusCode != http.StatusOK {
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			return nil, ErrPDHTTPRequestFailed.New("http status code: %d, msg: %s", resp.StatusCode, string(body))
 		}
-		return nil, err
+		return read(resp.Body)
```

Comment:
still need `resp.Body.Close()`?

Reply:
read is a generic function that has handled Close.
```
func read(stream io.ReadCloser) (*RegionsInfo, error) {
	defer stream.Close()
```